### PR TITLE
regex_replace multiline support

### LINKED ISSUE #61985: Multiline support for the regex_replace filter
##### SUMMARY
I think the `regex_replace` jinja filter should support `multiline` as the `regex_search` do.
Or maybe use a `flags` parameter to support any `re` flag.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
- `lib/ansible/plugins/filter/core.py`

##### ADDITIONAL INFORMATION
Example: comment a multiline variable.
```yaml
  val: "{{ val | regex_replace('^', '#', multiline=True) }}"
```

### PR DESCRIPTION
**Pull request by dr4Ke**
##### SUMMARY

This is a change to add a multiline parameter to the `regex_replace` filter.
Fixes #61985

##### ISSUE TYPE

- Docs Pull Request
- Feature Pull Request

##### COMPONENT NAME

regex_replace filter in lib/ansible/plugins/filter/core.py

##### ADDITIONAL INFORMATION

playbook:
```
- name: test multiline string replace
  hosts: localhost
  tasks:
  - name: set multiline var
    set_fact:
      var: |
           # This is a configuration block
           MyConf:
             value1: yes
             value2: no
  - name: print var
    debug:
      msg: "var: {{ var }}"
  - name: modify var multiline
    set_fact:
      monoline: "{{ var | regex_replace('^', '#DISABLE#') }}"
  - name: print monoline
    debug:
      msg: "monoline: {{ monoline }}"
  - name: modify var multiline
    set_fact:
      multiline: "{{ var | regex_replace('^[^$]', '#DISABLE#', multiline=True) }}"
  - name: print multiline
    debug:
      msg: "multiline: {{ multiline }}"
```
Execute:
```
$ ansible-playbook test_regex_replace_multiline.yml 

PLAY [test multiline string replace] *********************************************************************************

TASK [Gathering Facts] ***********************************************************************************************
ok: [localhost]

TASK [set multiline var] *********************************************************************************************
ok: [localhost]

TASK [print var] *****************************************************************************************************
ok: [localhost] => {
    "msg": "var: # This is a configuration block\nMyConf:\n  value1: yes\n  value2: no\n"
}

TASK [modify var multiline] ******************************************************************************************
ok: [localhost]

TASK [print monoline] ************************************************************************************************
ok: [localhost] => {
    "msg": "monoline: #DISABLE## This is a configuration block\nMyConf:\n  value1: yes\n  value2: no\n"
}

TASK [modify var multiline] ******************************************************************************************
ok: [localhost]

TASK [print multiline] ***********************************************************************************************
ok: [localhost] => {
    "msg": "multiline: #DISABLE# This is a configuration block\n#DISABLE#yConf:\n#DISABLE# value1: yes\n#DISABLE# value2: no\n"
}

PLAY RECAP ***********************************************************************************************************
localhost                  : ok=7    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```

### CODE REVIEW

**Comment by samdoran** [comment]
Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment) and add integration tests. See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

This could lead to a slippery slope of adding every single method/option of `re` as a parameter of this filter. Adding this one is ok, but we need to make sure we don't open the floodgates in the future.

**Comment by dr4Ke** [comment]
Just added the changelog fragment. Sorry for the delay.

File: changelogs/fragments/65051-regex-replace-multiline.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+- core - add multiline support for regex_replace filter
```
Comment (samdoran) [inline changelogs/fragments/65051-regex-replace-multiline.yaml]:
```suggestion
- regexp_replace filter - add multiline support for regex_replace filter (https://github.com/ansible/ansible/issues/61985)
```

**Comment by samdoran** [comment]
Please add integration tests for this new option. Existing filter integration tests are [here](https://github.com/ansible/ansible/tree/devel/test/integration/targets/filters).

**Review by samccann — APPROVED** [review]
Docs portion LGTM.

Can you please rebase? instructions are here if you need them - https://docs.ansible.com/ansible/latest/dev_guide/developing_rebasing.html

thanks!

**Comment by dr4Ke** [comment]
Just rebased on devel.

**Comment by dr4Ke** [comment]
Oops. I left a part of my work behind (the tests).

**Comment by dr4Ke** [comment]
Just rebased, correctly this time.

**Review by samdoran — APPROVED** [review]

**Pull request merged**